src/System.Console/tests/ReadAndWrite.cs

```diff
@@ -261,6 +259,30 @@ public static unsafe void ValidateConsoleEncoding(Encoding encoding)
     }
 
     [Fact]
+    // On the full framework it is not guaranteed to eat the preamble bytes
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+    public static unsafe void OutputEncodingPreamble()
+    {
+        Encoding curEncoding = Console.OutputEncoding;
+
+        try
+        {
+            Encoding encoding = Console.Out.Encoding;
+            // The primary purpose of ConsoleEncoding is to return an empty preamble.
+            Assert.Equal(Array.Empty<byte>(), encoding.GetPreamble());
```

Comment:
So Console.Out.Encoding always works correctly but Console.OutputEncoding doesn't?  Without looking at the implementation, I'd have expected those to be the same, unless something is changing one of them but not the other earlier in the process?